doc/configuration.md

```diff
@@ -0,0 +1,247 @@
+Configuration
+=============
+The Datadog tracer's configuration can be specified in multiple ways:
+
+- programmatically in C++ code via the `TracerOptions` object defined in
+  [datadog/opentracing.h][1],
+- process-wide via setting values for certain [environment variables][4],
+- [dynamically][3] by reading JSON-formatted text, as in done in the [nginx
+  plugin][2].
+
+Most options support all three methods of configuration.
+
+Environment variables override any corresponding configuration in
+`TracerOptions` or loaded from JSON.
+
+Options
+-------
+### Agent Host
+The name of the host at which the Datadog Agent can be contacted.
+
+- **TracerOptions member**: `std::string agent_host`
+- **JSON property**: `"agent_host"` _string_
+- **Environment variable**: `DD_AGENT_HOST`
+- **Default value**: `"localhost"`
+
+### Agent Port
+The port on which the Datadog Agent is listening.
+
+- **TracerOptions member**: `uint32_t agent_port`
+- **JSON property**: `"agent_port"` _integer_
+- **Environment variable**: `DD_TRACE_AGENT_PORT`
+- **Default value**: `8126`
+
+### Agent URL
+As an alternative to specifying a host and port separately, a URL may be
+specified indicating where the Datadog Agent can be contacted.  Both TCP
+and Unix domain sockets are supported.  For more information about using a
+Unix domain socket, see the [relevant example][5].
+
+The following forms are supported:
+
+- `http://host` (TCP)
+- `http://host:port` (TCP)
+- `https://host` (TCP; TLS is not used)
```

Comment:
🤷  My understanding is that now we don't support TLS when contacting the agent, because certs are a complicating factor. Supporting it would require agent changes, too. I think it's on your wish list.

I could skirt over the TLS point (I think that would be fair).  Also, I could remove `https` from the documentation. It could be supported as an undocumented extension (with the tricky caveat that TLS isn't actually used...).

What do you think?